cdap-ui/app/cdap/main.js

```diff
@@ -54,12 +54,39 @@ import { Theme } from 'services/ThemeHelper';
 import AuthRefresher from 'components/AuthRefresher';
 import ThemeWrapper from 'components/ThemeWrapper';
 import './globals';
+import { ApolloProvider } from 'react-apollo';
+import ApolloClient from 'apollo-boost';
+import { IntrospectionFragmentMatcher, InMemoryCache } from 'apollo-cache-inmemory';
+// See ./graphql/fragements/README.md
+import introspectionQueryResultData from '../../graphql/fragments/fragmentTypes.json';
 
 const Administration = Loadable({
   loader: () => import(/* webpackChunkName: "Administration" */ 'components/Administration'),
   loading: LoadingSVGCentered,
 });
 
+const fragmentMatcher = new IntrospectionFragmentMatcher({
+  introspectionQueryResultData,
+});
+
+const client = new ApolloClient({
+  uri: '/graphql',
+  cache: new InMemoryCache({ fragmentMatcher }),
+  request: async (operation) => {
+    let token = '';
+
+    if (cookie.load('CDAP_Auth_Token')) {
+      token = `Bearer ${cookie.load('CDAP_Auth_Token')}`;
+    }
+
+    operation.setContext({
+      headers: {
+        authorization: token,
+      },
+    });
```

Comment:
you should check whether security is enabled or not. If it's not, fetch the token and pass in the headers. otherwise, don't pass in headers at all

Reply:
Done